File: 557. reverse-string/main.cpp
#include <iostream>
#include "vector"

using namespace std;

class Solution {
public:
    string reverseWords(string s) {
        size_t front = 0;
        for(int i = 0; i<=s.length(); i++){
            if(s[i] == ' ' || i == s.length()){
                reverse(&s[front], &s[i]); // reverse the last traversed word
                front = i + 1; // 1 is for space
            }
        }
        return s;
    }
};
int main() {
    Solution sol;
    cout << sol.reverseWords(std::string ("Abhinandan Dubey"));
    return 0;
}

File: 557. reverse-string/main.py
def reverseWords(str):
    str1 = str.split(" ")
    str2 = []
    for word in str1:
        wdsplit = list(word)
        wdsplit = reversed(wdsplit)
        print wdsplit
        str2.append("".join(wdsplit))
    str2 = " ".join(str2)
    return str2

reverseWords("Abhinandan Dubey")